### DISCUSSION TITLE
Caching data at mobile app

### DISCUSSION THREAD

**Discussion by NaveedAhmadHematmal** [VDTS/CodeX.VaxineSolution#33] [Q&A]
How do you think we should cache data, and which data is important to cache in the app?
Should we cache all pages data or relevant and recent data?
Do we need to cache newly added data in a queue to ensure data loss prevention in slow internet connection or offline mode?

**Comment by YasinZahin** [comment]
د موبلايزر دخوا هره اضافه سوې ډيټا بايد cache سي ځکه هغه ټول مهم دي
او د والدينو باید فقط feedback برخه cache سي.

**Reply by SaeedaRasuly** [reply]
@NaveedAhmadHematmal In my point of view, the data that does not need to frequently change; the ones that are the same for all users and don't change often.  should be cached. E.g. Reports, static images..

**Reply by SaeedaRasuly** [reply]
Also data like user profile, historical data and static data

**Reply by SaeedaRasuly** [reply]
All the static content must be cached rather then dynamic

**Reply by NaveedAhmadHematmal** [reply]
@YasinZahin
We need to develop two types of caches.
Temporary caches: That caches for a limit time, when internet is available, it uploads and deletes from cache.
Offline caches: Saves data for offline mode.

Feedback page data drops in the first category.

**Reply by NaveedAhmadHematmal** [reply]
@SaeedaRasuly

Good point. We don't need to refresh static data everytime, like profile image. I will include it in the plan.

**Reply by SaeedaRasuly** [reply]
Yeah, as per my research most of static data is often cached but we can also cache dynamic data too

**Reply by NaveedAhmadHematmal** [reply]
> Yeah, as per my research most of static data is often cached but we can also cache dynamic data too

Yeah, I have mentioned above that we have two caches, one for the scenario you are mentioning and one for other data.
